src/tests/ftest/control/dmg_pool_query_ranks.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/python3
+"""
+  (C) Copyright 2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import time
+
+from control_test_base import ControlTestBase
+
+
+class DmgPoolQueryRanks(ControlTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test dmg query ranks enabled/disabled command.
+
+    Test Class Description:
+        Simple test to verify the pool query command ranks of dmg tool.
+
+    :avocado: recursive
+    """
+
+    def setUp(self):
+        """Set up for dmg pool query."""
+        super().setUp()
+
+        # Init the pool
+        self.add_pool(connect=False)
+
+    def test_pool_query_ranks_basic(self):
+        """Test the state of ranks with dmg pool query
+
+        Test Description:
+            Create a pool with some engines and check they are all enabled.
+
+        :avocado: tags=all,daily_regression
+        :avocado: tags=vm
+        :avocado: tags=dmg,pool_query,pool_query_ranks
```

Comment:
Minor - Could you also add the `control` feature tag?
```suggestion
        :avocado: tags=dmg,control,pool_query,pool_query_ranks
```

Reply:
- [ ] Add control to avocado tags